website/addons/s3/static/s3NodeConfig.js

```diff
@@ -89,33 +89,53 @@ ViewModel.prototype.updateBucketList = function(){
         });
 };
 
+ViewModel.prototype.isValidBucketName = function(bucketName, allowPeriods) {
```

Comment:
This doesn't need to be a method (one hint: no use of `this`); it can just be a function.